app/api.py

```diff
@@ -1,39 +1,39 @@
 """File defining all the routes for the module, to configure the router"""
+import glob
+import importlib
+import logging
+from typing import Iterable
 
 from fastapi import APIRouter
 
-from app.endpoints import (
-    admin,
-    advert,
-    amap,
-    associations,
-    auth,
-    booking,
-    calendar,
-    campaign,
-    cinema,
-    core,
-    groups,
-    loan,
-    notification,
-    raffle,
-    users,
-)
+from app.core import endpoints_core
+from app.core.auth import endpoints_auth
+from app.core.groups import endpoints_groups
+from app.core.module import Module
+from app.core.notification import endpoints_notification
+from app.core.users import endpoints_users
 
+hyperion_error_logger = logging.getLogger("hyperion.error")
+
+module_list: list[Module] = []
 api_router = APIRouter()
 
-api_router.include_router(admin.router)
-api_router.include_router(advert.router)
-api_router.include_router(amap.router)
-api_router.include_router(associations.router)
-api_router.include_router(auth.router)
-api_router.include_router(booking.router)
-api_router.include_router(calendar.router)
-api_router.include_router(campaign.router)
-api_router.include_router(cinema.router)
-api_router.include_router(core.router)
-api_router.include_router(groups.router)
-api_router.include_router(loan.router)
-api_router.include_router(notification.router)
-api_router.include_router(raffle.router)
-api_router.include_router(users.router)
+api_router.include_router(endpoints_auth.router)
+api_router.include_router(endpoints_core.router)
+api_router.include_router(endpoints_groups.router)
+api_router.include_router(endpoints_notification.router)
+api_router.include_router(endpoints_users.router)
+
+for endpoints_file in glob.glob("./app/modules/*/endpoints_*.py"):
+    endpoint_module = importlib.import_module(endpoints_file[2:-3].replace("/", "."))
+    if hasattr(endpoint_module, "module"):
+        modules = endpoint_module.module
+        if not isinstance(modules, Iterable):
+            modules = [modules]
+        module_list.extend(modules)
```

Comment:
I'm not sure if this is really a good idea to accept both an Iterable and a single Module

Reply:
Fixed, now we can only declare one module per file